src/Compilers/CSharp/Test/Semantic/Semantics/GenericConstraintsTests.cs

```diff
@@ -3629,5 +3629,307 @@ public struct YourStruct<T> where T : unmanaged
+    {
+        var arr = stackalloc MyStruct<int>[4];
+    }
+}
+";
+            CreateCompilation(code, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                    // (8,30): error CS8370: Feature 'unmanaged generic structs' is not available in C# 7.3. Please use language version 8.0 or greater.
+                    //         var arr = stackalloc MyStruct<int>[4];
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "MyStruct<int>").WithArguments("unmanaged generic structs", "8.0").WithLocation(8, 30)
+                );
+        }
+
+        [Fact]
+        public void GenericStructPointerFieldRequiresCSharp8()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+}
+
+public unsafe struct OtherStruct
+{
+    public MyStruct<int>* ms;
+}
+";
+            CreateCompilation(code, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                    // (9,12): error CS8370: Feature 'unmanaged generic structs' is not available in C# 7.3. Please use language version 8.0 or greater.
+                    //     public MyStruct<int>* ms;
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "MyStruct<int>*").WithArguments("unmanaged generic structs", "8.0").WithLocation(9, 12)
+                );
+        }
+
+        [Fact]
+        public void GenericRefStructAddressOf()
+        {
+            var code = @"
+public ref struct MyStruct<T>
+{
+    public T field;
+}
+
+public class MyClass
+{
+    public static unsafe void Main()
+    {
+        var ms = new MyStruct<int>() { field = 42 };
+        var ptr = &ms;
+        System.Console.Write(ptr->field);
+    }
+}
+";
+
+            CompileAndVerify(code,
+                options: TestOptions.UnsafeReleaseExe,
+                verify: Verification.Skipped,
+                expectedOutput: "42");
+        }
+
+        [Fact]
+        public void GenericStructFixedStatement()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+}
+
+public class MyClass
+{
+    public MyStruct<int> ms;
+    public static unsafe void Main()
+    {
+        var c = new MyClass();
+        c.ms.field = 42;
+        fixed (MyStruct<int>* ptr = &c.ms)
+        {
+            System.Console.Write(ptr->field);
+        }
+    }
+}
+";
+
+            CompileAndVerify(code,
+                options: TestOptions.UnsafeReleaseExe,
+                verify: Verification.Skipped,
+                expectedOutput: "42");
+        }
+
+        [Fact]
+        public void GenericStructLocalFixedStatement()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+}
+
+public class MyClass
+{
+    public static unsafe void Main()
+    {
+        var ms = new MyStruct<int>();
+        fixed (int* ptr = &ms.field)
```

Comment:
Do you have a `fixed` test where the struct isn't already in a fixed position?

Reply:
See GenericStructFixedStatement above.